tests/cmdline/params/options/test_interactive.py

```diff
@@ -95,6 +110,20 @@ def test_callback_prompt_twice(self):
         self.assertIn(expected_3, lines[9])
         self.assertIn(expected_4, lines[12])
 
+    def test_callback_prompt_only_once(self):
+        """
+        scenario: using InteractiveOption with type=float and callback that echos an additional message
+        behaviour: the callback should be called at most once per prompt
+        """
+        cmd = self.simple_command(type=float, callback=validate_positive_number_with_echo)
+        runner = CliRunner()
+        result = runner.invoke(cmd, [], input='string\n-1\n-1\n1\n')
+        self.assertIsNone(result.exception)
+        lines = result.output.split('\n')
+        #The callback should be called once per prompt
+        #where type conversion was successful
+        self.assertEqual(lines.count('Validating Number'), 3)
```

Comment:
With the change of including the value being passed in the echo string, I think we can make the test more robust and make the changes on false positives smaller. Since the echo should be the only output (right?) we should also be able to check the entire output verbatim. So I would suggest something like the following:

```suggestion
        self.assertEqual(result.output, 'Validating string\nValidating -1\nValidating 1\n')
```

Reply:
I agree that the test can be made more robust by including the value in the echo, but the change will not work as is, since for example ``'Validating string'`` will not appear in the output. This value will fail when it is converted to a float so the callback will never be called.

Reply:
Ah I see, because it would already fail the validation at the type which is expecting a float. Well, by accident, I think that is actually good to keep the `string` value in, as it tests that that _doesn't_ hit the callback as it shouldn't. So maybe all you have to change in my suggestion is just to remove `Validate string\n` from the expected output

Reply:
I will probably need to check specific lines of the output, since there is more stuff in ``result.output`` besides the ``'Validating {value}'`` lines like the lines for entering the values for example but in principle yes